astropy/io/fits/tests/test_convenience.py

```diff
@@ -494,3 +495,17 @@ def test_getdata_ext_not_given_nodata_noext(self):
             IndexError, match="No data in Primary HDU and no extension HDU found."
         ):
             fits.getdata(buf)
+
+    def test_writeto_stdout(self):
+        # see https://github.com/astropy/astropy/issues/3427
+        fits.writeto(sys.stdout, data=np.array([1, 2]))
+
+    def test_HDU_writeto_stdout(self):
+        # see https://github.com/astropy/astropy/issues/3427
+        hdu = fits.PrimaryHDU()
+        hdu.writeto(sys.stdout)
+
+    def test_HDUList_writeto_stdout(self):
+        # see https://github.com/astropy/astropy/issues/3427
+        hdul = fits.HDUList([fits.PrimaryHDU()])
+        hdul.writeto(sys.stdout)
```

Comment:
Let's shorten a bit the test (it already difficult to triage them already :)). There are probably other `.writeto` tests elsewhere, not sure what's the best location. Probably `test_core`.

```suggestion
    def test_writeto_stdout(self):
        # see https://github.com/astropy/astropy/issues/3427
        fits.writeto(sys.stdout, data=np.array([1, 2]))

        hdul = fits.HDUList([fits.PrimaryHDU()])
        hdul.writeto(sys.stdout)
```

Reply:
Thank you ! I've moved the tests to hopefully more appropriate locations and dropped one (`test_HDU_writeto_stdout`), however I noticed that the unified version you suggested was raising some warnings (at least two), that I assume are artefacts of the testing context. Anyway, I figured it might actually be more maintainable to keep them separated.